examples/api/lib/material/selection_area/selection_area.1.dart

```diff
@@ -0,0 +1,372 @@
+            }
+            if (selectionDetails.status != SelectionStatus.uncollapsed
+               || !selectionDetails.selectionFinalized) {
+              return;
+            }
+            if (selectionAreaKey.currentState == null
+                || !selectionAreaKey.currentState!.mounted
+                || selectionAreaKey.currentState!.selectableRegion.contextMenuAnchors.secondaryAnchor == null) {
+              return;
+            }
+            _menuController.show(
+              context: context,
+              contextMenuBuilder: (BuildContext context) {
+                return TapRegion(
+                  onTapOutside: (PointerDownEvent event) {
+                    if (_menuController.isShown) {
+                      ContextMenuController.removeAny();
+                    }
+                  },
+                  child: AdaptiveTextSelectionToolbar.buttonItems(
+                    buttonItems: <ContextMenuButtonItem>[
+                      ContextMenuButtonItem(
+                        onPressed: () {
+                          ContextMenuController.removeAny();
+                          _colorSelectionRed(
+                            selectionDetails,
+                            dataMap: dataSourceMap,
+                            coloringChildSpan: false,
+                          );
+                          selectionAreaKey.currentState!.selectableRegion.clearSelection();
+                        },
+                        label: 'Color Text Red',
+                      ),
+                    ],
+                    anchors: TextSelectionToolbarAnchors(primaryAnchor: selectionAreaKey.currentState!.selectableRegion.contextMenuAnchors.secondaryAnchor!),
+                  ),
+                );
+              },
+            );
+          },
+          child: Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                for (final MapEntry<_GlobalSpanRange, TextSpan> entry in dataSourceMap.entries)
+                  Text.rich(
+                    entry.value,
+                  ),
+              ],
+            ),
+          ),
+        ),
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () {
+          setState(() {
+            // Resets the state to the origin data.
+            for (final MapEntry<_GlobalSpanRange, TextSpan> entry in originSourceData.entries) {
+              dataSourceMap[entry.key] = entry.value;
+            }
+            for (final MapEntry<_GlobalSpanRange, TextSpan> entry in originBulletSourceData.entries) {
+              bulletSourceMap[entry.key] = entry.value;
+            }
+            for (final MapEntry<_GlobalSpanRange, TextSpan> entry in originEmphasizedSourceData.entries) {
+              emphasizedTextMap[entry.key] = entry.value;
+            }
+          });
+        },
+        child: const Icon(Icons.undo),
+      ),
+    );
+  }
+}
```

Comment:
I talked offline with @Renzo-Olivares but leaving this dartpad link here as an alternative way of going about these examples. Not sure if there's anything useful in it now. https://dartpad.dev/?id=97bd3f3cec6f90e3ca3791f085ac3b2c